Products/Zuul/routers/device.py

```diff
@@ -948,7 +949,8 @@ def setCollector(self, uids, collector, hashcheck, uid=None, ranges=(),
                 return DirectResponse.succeed('Changed collector to %s for %s devices.' %
                                       (collector, len(uids)))
         except Exception, e:
-            log.exception(e)
+            log.debug(traceback.format_exc())
+            log.error(e)
```

Comment:
For a slightly more efficient (i.e. avoid formatting the traceback which happens regardless of whether it's logged) and keeping the error to a single entry in the log, I suggest:
```
    if log.isEnabledFor(logging.DEBUG):
        log.exception(e)
    else:
        log.error(e)
```
And it also doesn't log a traceback as a DEBUG log entry.